black.py

```diff
@@ -147,18 +162,28 @@ def format_file_in_place(src: Path, line_length: int, fast: bool) -> bool:
     return True
 
 
+def format_stdin_to_stdout(line_length: int, fast: bool) -> bool:
+    """Format file on stdin and pipe output to stdout. Return True if changed."""
+    try:
+        contents, _ = format_file(sys.stdin, line_length=line_length, fast=fast)
+    except NothingChanged:
+        return False
+
+    sys.stdout.write(contents)
+    return True
+
+
 def format_file(
-    src: Path, line_length: int, fast: bool
+    src_buffer: Type[io.TextIOBase], line_length: int, fast: bool
```

Comment:
I work on typing, too, so I'm wondering what made you choose `Type[]` here and why you chose `io.TextIOBase` specifically. Both choices are wrong here and I will explain why.

1. `Type[]` is to signify the caller is going to pass a *class*, not an *object*. Here we want a regular object.
2. `io.TextIOBase` is overly abstract, it doesn't have `name` for example.

Instead, you would use `typing.TextIO` to describe what you wanted to achieve. Don't worry if you didn't get this until now, this is very new tech. So I really wonder about your first experiences here. I should also add `mypy` to CI to make stuff like this more apparent :-)

# Practical suggestions

1. Rename `src_buffer` back to `src`. It's an abstract enough name and the function's name suggests we want a file-like object anyway.
2. Change the docstring to say "an open file" instead of "a file" to make this more obvious.
3. Change the type to `TextIO` (import it from `typing` instead of `Type` above).

Reply:
Ah, I clearly misunderstood the purpose of `Type`. I thught it was to specify an object of a class *or subclass* with the given type. Not running mypy much led to me never finding out I was getting this wrong! :laughing: Will have to read throgh `typing`'s docs a bit more thoroughly.